search-provider/eks-discovery-feed-provider.c

```diff
@@ -574,9 +591,11 @@ handle_artwork_card_descriptions (EksDiscoveryFeedDatabaseContentProvider *skele
                                                 NULL),
                                   get_day_of_year (),
                                   DAYS_IN_YEAR,
+                                  invocation,
                                   self->cancellable,
                                   artwork_card_descriptions_cb,
-                                  discovery_feed_query_state_new (invocation, self));
+                                  discovery_feed_query_state_new (invocation, self),
+                                  (GDestroyNotify) discovery_feed_query_state_free);
```

Comment:
For this and the following uses of `discovery_feed_query_state_free()`: was this not freed before? or do you have to remove a usage of the free function somewhere else?

Reply:
If I remember correctly - this was not freed before and I had meant to put it in a separate commit. I'll revisit this to see if that's feasible.